parsing/cLexer.mli

```diff
@@ -8,8 +8,32 @@
 (*         *     (see LICENSE file for the text of the license)         *)
 (************************************************************************)
 
+(** When one registers a keyword he can declare it starts a quotation.
+  In particular using QUOTATION("name:") in a grammar rule
+  declares "name:" as a keyword and the token QUOTATION is
+  matched whenever the keyword is followed by an identifier or a
+  parenthesized text. Eg
+
+    constr:x
+    string:[....]
+    ltac:(....)
+    ltac:{....}
+
+  The delimiter is made of 1 or more occurrences of the same parenthesis,
+  eg ((.....)) or [[[[....]]]]. The idea being that if the text happens to
+  contain the closing delimiter, one can make the delimiter longer and avoid
+  confusion (no escaping). Eg
+
+    string:[[ .. ']' .. ]]
+
+
+  Nesting the delimiter is allowed, eg ((..((...))..)) is OK.
+
+  Keywords don't need to end in ':' *)
+type starts_quotation = NoQuotation | Quotation
+
 (** This should be functional but it is not due to the interface *)
-val add_keyword : string -> unit
+val add_keyword : ?quotation:starts_quotation -> string -> unit
```

Comment:
Maybe it would be clearer to have two functions ? `add_keyword` and `add_quotation` ?

Reply:
I thought about that, but in the current model a quotation is always initiated by a keyword. So the API would look like
```
val add_keyword : string -> unit
val add_keyword_for_quotation : string -> unit
```
that is pretty much what we have in the patch...